scipy/signal/tests/test_signaltools.py

```diff
@@ -156,40 +156,55 @@ def test_invalid_shapes(self):
         for mode in ['full', 'valid', 'same']:
-            for _, dtype_list in np.sctypes.items():
-                for dtype in dtype_list:
-                    if dtype == np.void or dtype == str:
-                        continue
-                    np.random.seed(42)
-                    x = (0.5 + np.random.rand(100)).astype(dtype)
-                    h = (0.5 + np.random.rand(50)).astype(dtype)
-
-                    if x.dtype.kind not in 'buifc' \
-                            or np.dtype(dtype).name in {'complex256', 'complex192'}:
-                        assert_equal(choose_conv_method(x, h, mode=mode), 'direct')
-                        # continues the for-loop; it will throw an error
-                        # because fftconv doesn't support the dtype
-                        continue
-
-                    if x.dtype.kind != 'b':
-                        x += 1
-                        h += 1
-
-                    if x.dtype.kind == 'c':
-                        x += 1j * np.random.rand(*x.shape).astype(dtype)
-                        h += 1j * np.random.rand(*h.shape).astype(dtype)
-
-                    results = {'direct':convolve(x, h, mode=mode,
-                                                 method='direct'),
-                               'fft':convolve(x, h, mode=mode, method='fft')}
-
-                    rtol = {'rtol': 2.0e-3} if dtype in {np.complex64,
-                                                         np.float32} else {}
-                    if isinstance(results['direct'], np.ndarray):
-                        assert_allclose(results['fft'], results['direct'], **rtol)
-                        assert_equal(results['direct'].dtype, results['fft'].dtype)
+            for dtype in types:
+                if dtype == np.void or dtype == str:
+                    continue
+                np.random.seed(42)
+                x = (0.5 + np.random.rand(100)).astype(dtype)
+                h = (0.5 + np.random.rand(50)).astype(dtype)
+
+                if x.dtype.kind not in 'buifc' \
+                        or np.dtype(dtype).name in {'complex256', 'complex192'}:
+                    assert_equal(choose_conv_method(x, h, mode=mode), 'direct')
+                    continue
+
+                if x.dtype.kind != 'b':
+                    x += 1
+                    h += 1
+
+                if x.dtype.kind == 'c':
+                    x += 1j * np.random.rand(*x.shape).astype(dtype)
+                    h += 1j * np.random.rand(*h.shape).astype(dtype)
+
+                results = {key: convolve(x, h, mode=mode, method=key)
+                           for key in ['fft', 'direct']}
+
+                rtol = {'rtol': 2.0e-3} if dtype in {np.dtype('complex64'),
+                                                     np.dtype('float32')} else {}
+                if isinstance(results['direct'], np.ndarray):
+                    assert_allclose(results['fft'], results['direct'], **rtol)
```

Comment:
What value would that add? This dtype check is less complete than the other checks (this check only looks at same dtypes for 1D; other checks look at all dtype combinations for 1D).